tests/model_formsets_regress/tests.py

```diff
@@ -279,6 +279,12 @@ def test_initial_data(self):
         self.assertEqual(formset.extra_forms[0].initial["data"], 41)
         self.assertIn('value="42"', formset.extra_forms[1].as_p())
 
+    def test_use_base_model_form_set_in_inlineformset_factory(self):
+        FormSet = inlineformset_factory(User, UserSite, formset=BaseModelFormSet, fields="__all__")
+        form_set = FormSet()
+        with self.assertNumQueries(1):
+            form_set.save()
+
```

Comment:
This is not a proper regression test, it works without the patch. I would add a test with `BaseModelFormSet` used by `formset_factory`, e.g.
```diff
diff --git a/tests/model_formsets/tests.py b/tests/model_formsets/tests.py
index a456e1a65e..87ea7bb6ea 100644
--- a/tests/model_formsets/tests.py
+++ b/tests/model_formsets/tests.py
@@ -12,6 +12,7 @@ from django.forms.models import (
     inlineformset_factory,
     modelformset_factory,
 )
+from django.forms.formsets import formset_factory
 from django.http import QueryDict
 from django.test import TestCase, skipUnlessDBFeature
 
@@ -2066,6 +2067,32 @@ class ModelFormsetTest(TestCase):
         formset.save()
         self.assertCountEqual(Author.objects.all(), [charles, walt])
 
+    def test_formset_factory_basemodelformset(self):
+        charles = Author.objects.create(name="Charles Baudelaire")
+
+        class AuthorForm(forms.ModelForm):
+            class Meta:
+                model = Author
+                fields = "__all__"
+
+        class BaseAuthorFormSet(BaseModelFormSet):
+            def __init__(self, *args, **kwargs):
+                self.model = Author
+                super().__init__(*args, **kwargs)
+
+        AuthorFormSet = formset_factory(AuthorForm, formset=BaseAuthorFormSet)
+        data = {
+            "form-TOTAL_FORMS": "2",
+            "form-INITIAL_FORMS": "0",
+            "form-MAX_NUM_FORMS": "0",
+            "form-0-name": "Arthur Rimbaud",
+            "form-1-name": "Walt Whitman",
+        }
+        formset = AuthorFormSet(data)
+        self.assertIs(formset.is_valid(), True)
+        formset.save()
+        self.assertEqual(Author.objects.count(), 3)
+
 
 class TestModelFormsetOverridesTroughFormMeta(TestCase):
     def test_modelformset_factory_widgets(self):
```